dsgrid/utils/spark.py

```diff
@@ -85,36 +91,38 @@ def read_dataframe(filename, cache=False, require_unique=None, read_with_spark=T
         Raised if a require_unique column has duplicate values.
 
     """
-    filename = Path(filename)
     func = _read_with_spark if read_with_spark else _read_natively
-    df = func(filename)
+    df = func(str(filename))
     _post_process_dataframe(df, cache=cache, require_unique=require_unique)
     return df
 
 
 def _read_with_spark(filename):
     spark = SparkSession.getActiveSession()
-    path = str(filename)
-    if filename.suffix == ".csv":
-        df = spark.read.csv(path, inferSchema=True, header=True)
+    if filename.endswith(".csv"):
```

Comment:
Does `filename` here need to be str for spark.read.csv()? If not, use `Path(filename).suffix`?